coral/src/services/test-utils/render-with-wrappers.tsx

```diff
@@ -1,148 +1,100 @@
-    ),
-    ...options,
-  });
-}
-
-function withMemoryRouter({
-  ui,
-  customRoutePath,
-  options,
-}: {
-  ui: ReactElement;
-  customRoutePath?: string;
-  options?: RenderOptions;
-}) {
-  return render(ui, {
-    wrapper: ({ children }) => (
-      <MemoryRouter initialEntries={[customRoutePath || ""]}>
-        {children}
-      </MemoryRouter>
-    ),
-    ...options,
-  });
-}
-
-function withMemoryRouterAndQueryClient({
-  ui,
-  customRoutePath,
-  options,
-}: {
-  ui: ReactElement;
+/***
+ * browserRouter: Only use if needed, e.g., for testing URL states (i.e., search params)
+ * in general, prefer memoryRouter for tests.
+ */
+type RenderConfig = {
+  queryClient?: boolean;
+  memoryRouter?: boolean;
+  browserRouter?: boolean;
+  aquariumContext?: boolean;
   customRoutePath?: string;
-  options?: RenderOptions;
-}) {
-  const queryClient = getQueryClientForTests();
-  return render(ui, {
-    wrapper: ({ children }) => (
-      <QueryClientProvider client={queryClient}>
-        <MemoryRouter initialEntries={[customRoutePath || ""]}>
-          {children}
-        </MemoryRouter>
-      </QueryClientProvider>
-    ),
-    ...options,
-  });
-}
-
-// Only use if BrowserRouter is really needed
-// prefer MemoryRouter for tests.
-function withBrowserRouterAndQueryClient({
-  ui,
-  options,
-}: {
-  ui: ReactElement;
-  options?: RenderOptions;
-}) {
-  const queryClient = getQueryClientForTests();
-  return render(ui, {
-    wrapper: ({ children }) => (
-      <QueryClientProvider client={queryClient}>
-        <BrowserRouter>{children}</BrowserRouter>
-      </QueryClientProvider>
-    ),
-    ...options,
-  });
-}
-
-// To be used for testing URL states (ie, search params)
-function withBrowserRouter({
-  ui,
-  options,
-}: {
-  ui: ReactElement;
-  options?: RenderOptions;
-}) {
-  return render(ui, {
-    wrapper: ({ children }) => <BrowserRouter>{children}</BrowserRouter>,
-    ...options,
-  });
-}
-
-type ValidOptions =
-  | "queryClient"
-  | "memoryRouter"
-  | "browserRouter"
-  | "customRoutePath";
+};
 
-type CustomRenderOption = {
-  [K in ValidOptions]?: K extends
-    | "queryClient"
-    | "memoryRouter"
-    | "browserRouter"
-    ? boolean
-    : string;
+type WrapperProps = {
+  children?: ReactNode;
 };
 
-function customRender(
+function createWrapper(
   ui: ReactElement,
-  renderWith: CustomRenderOption,
+  config: RenderConfig,
   options?: RenderOptions
 ) {
-  if (renderWith?.queryClient && renderWith?.memoryRouter) {
-    return withMemoryRouterAndQueryClient({
-      ui,
-      customRoutePath: renderWith?.customRoutePath,
-      options,
-    });
+  const {
+    queryClient,
+    memoryRouter,
+    browserRouter,
+    aquariumContext,
+    customRoutePath,
+  } = config;
+
+  const wrappers: React.FC<WrapperProps>[] = [];
+
+  if (!queryClient && !memoryRouter && !browserRouter) {
+    console.error(
+      "You have not passed any renderWith options, which returns the non-custom render."
```

Comment:
Maybe this error could be more explicit:

```
"You have not passed any renderWith options, which returns the non-custom render. Use the default render from @testing-library/react instead."
```

Alternatively, if options are required, then they should not be an optional argument to `customRender`, but required ?

Reply:
Ah, good idea! I tried around a bit and found added a utility type, so this can be handled by TS!

Reply:
We now have TypeErrors if the config option is empty or not there at all:

<img width="744" alt="Screenshot 2024-01-08 at 09 57 25" src="https://github.com/Aiven-Open/klaw/assets/943800/3636953e-0dde-48da-96cc-eb084d2e43a7">

<img width="930" alt="Screenshot 2024-01-08 at 09 57 16" src="https://github.com/Aiven-Open/klaw/assets/943800/3d2fcf34-3602-45dd-b979-49af3396afaf">

Reply:
Nice ^^ Thank you!